packages/server-wallet/src/objectives/objective-manager.ts

```diff
@@ -54,87 +66,58 @@ export class ObjectiveManager {
 
     while (attemptAnotherProtocolStep) {
       await this.store.lockApp(channelToLock, async tx => {
-        let executeProtocol: () => ChannelState.ProtocolResult;
-        let protocolState: OpenChannelProtocol.ProtocolState | CloseChannelProtocol.ProtocolState;
+        let nextAction: ChannelState.ProtocolResult;
+        let protocolState: SupportedProtocolState;
 
         if (objective.type === 'OpenChannel') {
-          protocolState = await OpenChannelProtocol.getOpenChannelProtocolState(
+          protocolState = await getOpenChannelProtocolState(
             this.store,
             objective.data.targetChannelId,
             tx
           );
-          executeProtocol = () =>
-            OpenChannelProtocol.protocol(protocolState as OpenChannelProtocol.ProtocolState);
+          nextAction = recordFunctionMetrics(
+            openChannelProtocol(protocolState as OpenChannelProtocolState),
```

Comment:
if `protocolState` is the return of `getOpenChannelProtocolState`, we shouldn't need to cast it on L79 (or anywhere).

Reply:
Yeah, I looked at this. Unfortunately line 70 explicitly declares `protocolState` to be the union of the two possible return types, so we need to cast. It needs to do this because it's used later in the function. This issue will go away as I continue to refactor though.